django_netjsonconfig/base/template.py

```diff
@@ -100,5 +103,21 @@ def clean(self, *args, **kwargs):
         if self.type == 'vpn' and not self.config:
             self.config = self.vpn.auto_client(auto_cert=self.auto_cert)
 
+    def clone(self, user):
+        clone = copy(self)
+        clone.name = self.name + " (CLONE)"
+        clone._state.adding = True
+        clone.pk = None
+        clone.full_clean()
+        clone.save()
+        ct = ContentType.objects.get_for_model(AbstractTemplate)
```

Comment:
`AbstractTemplate` is not a concrete template. You should be able to replace `AbstractTemplate` with `self.__class__`.